.github/workflows/formatting.yml

```diff
@@ -0,0 +1,39 @@
+name: Format code
+
+on: [push]
+
+jobs:
+  build:
+    strategy:
+      fail-fast: false
+      matrix:
+        python-version: ["3.9"]
+        os: [ubuntu-latest]
+    runs-on: ${{ matrix.os }}
+    steps:
+      - uses: actions/checkout@v2
+        with:
+          fetch-depth: 0
+
+      - name: Set up Python
+        uses: actions/setup-python@v1
+        with:
+          python-version: ${{ matrix.python-version }}
+
+      - name: Run Poetry image
+        uses: abatilo/actions-poetry@v2.3.0
+        with:
+          poetry-version: 1.7.1
+
+      - name: Install Dependencies
+        run: poetry install
+
+      - name: Autoformat code if the check fails
+        run: |
+          poetry run isort .
+          poetry run black .
+          git config --global user.name '${{ github.actor }}'
+          git config --global user.email '${{ github.actor }}@users.noreply.github.com'
+          git remote set-url origin https://x-access-token:${{ secrets.GITHUB_TOKEN }}@github.com/$GITHUB_REPOSITORY
```

Comment:
Did you generate these tokens yourself or are they generated by a general deltares account?

Reply:
No, I did not generate these myself, they are provided within the Deltares organization.

Reply:
nice!